cmp/grid/GridModel.js

```diff
@@ -72,14 +81,20 @@ export class GridModel {
         emptyText = null,
         sortBy = [],
         groupBy = null,
+        exportFilename = 'export',
+        exportFiletype,
+        enableExport = true,
         enableColChooser = false,
         stateModel = null,
         contextMenuFn = () => this.defaultContextMenu()
     }) {
         this.store = store;
         this.columns = columns;
-        this.contextMenuFn = contextMenuFn;
         this.emptyText = emptyText;
+        this.exportFilename = exportFilename;
+        this.exportFiletype = ['excel', 'excelTable', 'csv'].includes(exportFiletype) ? exportFiletype : 'excelTable';
+        this.enableExport = enableExport;
```

Comment:
Why not simply throw if the filetype is not acceptable, rather than quietly ignoring and setting to excelTable?